### ISSUE TITLE
support weighting loss functions for cost-sensitive learning

### ISSUE THREAD

**Issue by shane-huang** [intel-analytics/analytics-zoo#1057]
We need a way to support weighting loss functions for `cost-sensitive learning`. This means we are able to tell the model to "pay more attention" to samples from an under-represented class, or the model could be penaltized harder for misclassifying specific samples so that we can ensure the correctness of "important" samples. 

keras uses ```class_weight``` and ```sample_weights```  for weighting the loss function (see details in https://keras.io/models/sequential/#fit). "Sample_weights" looks not infeasible for big data. We need another way.
